apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.js

```diff
@@ -8,6 +8,99 @@ window.PLOrderBlocks = function (uuid, options) {
 
   let optionsElementId = '#order-blocks-options-' + uuid;
   let dropzoneElementId = '#order-blocks-dropzone-' + uuid;
+  let pressedKeys = {};
+
+  function initializeKeyboardHandling(optionsElementId, dropzoneElementId) {
+    function inDropzone(block) {
+      let parentArea = block.closest('.pl-order-blocks-connected-sortable');
+      return parentArea.classList.contains('dropzone');
+    }
+    function getIndentation(block) {
+      return Math.round(parseInt(block.style.marginLeft.replace('px', '') / TABWIDTH));
+    }
+    let fullContainer = document.querySelector('.pl-order-blocks-question-' + uuid);
+    let blocks = fullContainer.querySelectorAll('*.pl-order-block');
+
+    for (let block of blocks) {
+      block.addEventListener('click', () => {
+        block.focus();
+      });
+
+      block.addEventListener('blur', () => {
+        if (block.classList.contains('selected')) {
+          block.classList.remove('selected');
+        }
+      });
+
+      block.addEventListener('keydown', (ev) => {
+        pressedKeys[ev.key] = true;
+        if (!block.classList.contains('selected')) {
+          if (ev.key === 'Enter') {
+            block.classList.add('selected');
+          }
+        } else {
+          ev.preventDefault();
+          switch (ev.key) {
+            case 'ArrowDown':
+              if (block.nextElementSibling) {
+                block.nextElementSibling.insertAdjacentElement('afterend', block);
+              }
+              break;
+            case 'ArrowUp':
+              if (block.previousElementSibling) {
+                block.previousElementSibling.insertAdjacentElement('beforebegin', block);
+              }
+              break;
+            case 'Delete':
+            case 'Backspace':
+              if (inDropzone(block)) {
+                block.style.marginLeft = '0px';
+                if (block.hasAttribute('data-distractor-bin')) {
+                  let distractorBinId = block.getAttribute('data-distractor-bin');
+                  let distractorBin = fullContainer
+                    .querySelector(
+                      '.pl-order-blocks-pairing-indicator[data-distractor-bin=' +
+                        distractorBinId +
+                        ']',
+                    )
+                    .querySelector('.inner-list');
+                  distractorBin.insertAdjacentElement('afterbegin', block);
```

Comment:
You should be able to replace at least some (if not all) of this code by calling the existing `correctPairing` function.

Reply:
I'm not sure what should be passed in as the `ui` argument for `correctPairing`, so right now I can't get this working. Any suggestions?